wgpu-hal/src/gles/device.rs

```diff
@@ -48,7 +48,12 @@ impl CompilationContext<'_> {
 
         for (name, mapping) in reflection_info.texture_mapping {
             let var = &module.global_variables[mapping.texture];
-            let register = if var.storage_access.is_empty() {
+            let empty_storage_access = if let naga::StorageClass::Storage { access } = var.class {
```

Comment:
This needs to be rewritten now. We are checking whether this image is a texture or a storage image. The storage class is never going to be `StorageClass::Storage` - this is used for buffers, not textures. So we need to do something like this:
```rust
let register = match module.types[var.ty].inner {
  naga::TypeInner::Image { class: naga::ImageClass::Storage { .. } } => super::BindingRegister::Images,
  _ => super::BindingRegister::Textures,
};
```